MySaver/Views/StoresPage.xaml.cs

```diff
@@ -7,15 +7,42 @@ public partial class StoresPage : ContentPage
     private StoresViewModel viewModel;
     public StoresPage(StoresViewModel viewModel)
     {
-        InitializeComponent();
+        InitializeComponent();       
         BindingContext = viewModel;
         this.viewModel = viewModel;
     }
-
+    
     protected async override void OnAppearing()
     {
         base.OnAppearing();
         await viewModel.UpdateStoresAsync();
     }
+    
+    public void ShopsFilterByName(object sender, TextChangedEventArgs e)
+    {
+       
+        var searchTerm = e.NewTextValue;
+
+        if (string.IsNullOrWhiteSpace(searchTerm))
+        {
+            searchTerm = string.Empty;
+        }
+
+        searchTerm = searchTerm.ToLowerInvariant();
 
+        var filteredItems = viewModel.Stores.Where(Store => Store.Name.ToLowerInvariant().Contains(searchTerm)).ToList();
```

Comment:
This line of code is a bit long, would be easier to read if it was split to 2 lines instead.